app/tasksmodule/datachecks.py

```diff
@@ -0,0 +1,178 @@
+
+"""Tasks to check data.
+
+These tasks don't modify data, they just double-check things that are already
+expected to be true.
+"""
+
+import datetime
+import time
+
+import django.http
+from google.appengine import runtime
+from google.appengine.api import datastore_errors
+from google.appengine.api import taskqueue
+
+import tasksmodule.base
+import model
+import utils
+import views.base
+
+
+class DatacheckException(Exception):
+    """An exception raised when a datacheck tasks finds a problem."""
+    pass
+
+
+class DatachecksBaseTask(tasksmodule.base.PerRepoTaskBaseView):
+
+    def get_params(self):
+        return views.base.read_params(
+            super(DatachecksBaseTask, self).get_params(),
+            self.request,
+            post_params={'cursor': utils.strip})
+
+    def schedule_task(self, repo, **kwargs):
+        name = '%s-%s-%s' % (repo, self.BASE_NAME, int(time.time()*1000))
+        path = self.build_absolute_path('/%s/tasks/%s' % (repo, self.TASK_PATH))
+        cursor = kwargs.get('cursor', '')
+        # TODO(nworden): figure out why setting task_retry_limit isn't working
+        retry_options = taskqueue.taskqueue.TaskRetryOptions(task_retry_limit=1)
+        taskqueue.add(name=name, method='POST', url=path,
+                      queue_name='datachecks', retry_options=retry_options,
+                      params={'cursor': cursor})
+
+    def alert(self, msg):
+        """Alerts developers of an error.
+
+        For now, we just raise an exception. It will show up in Stackdriver.
+        """
+        raise DatacheckException(msg)
+
+
+class PersonDataValidityCheckTask(DatachecksBaseTask):
+    """Checks that person records have valid data."""
+
+    BASE_NAME = 'person_data_validity_check'
+    TASK_PATH = 'check_person_data_validity'
+
+    def _check_person(self, person):
+        if not person.entry_date:
+            self.alert('A person record is missing an entry_date value.')
+        if not person.original_creation_date:
+            self.alert(
+                'A person record is missing an original_creation_date value.')
+        if (person.author_email and
+                not utils.validate_email(person.author_email)):
+            self.alert('A person record has an invalid author_email value.')
+
+    def post(self, request, *args, **kwargs):
+        del request, args, kwargs  # unused
+        q = model.Person.all(filter_expired=False).filter(
+            'repo =', self.env.repo)
+        cursor = self.params.get('cursor', '')
+        if cursor:
+            q.with_cursor(cursor)
+        try:
+            for person in q:
+                next_cursor = q.cursor()
+                self._check_person(person)
+                cursor = next_cursor
+        except runtime.DeadlineExceededError:
+            self._schedule_task(self.env.repo, cursor)
+        except datastore_errors.Timeout:
+            self._schedule_task(self.env.repo, cursor)
+        return django.http.HttpResponse('')
```

Comment:
maybe add something like 'success' to the response?

Reply:
We actually don't know if it's successful, because it might have overrun the limit and be a continuing task. Since it's meant to be run as a cron job I'm not too concerned about making these responses meaningful.